lib/sdf-core/src/generate_template.rs

```diff
@@ -0,0 +1,204 @@
+        template_components.push(template_component);
+    }
+    for component in template_components.iter_mut() {
+        for attribute in component.attributes.iter_mut() {
+            if let AttributeSource::Subscription(sub_value) = &mut attribute.value {
+                let c_ulid = Ulid::from_string(&sub_value.component)?;
+                let c_id: ComponentId = ComponentId::from_raw_id(c_ulid.into());
+                if let Some(variable_name) = component_id_to_variable_name.get(&c_id) {
+                    sub_value.variable = Some(variable_name.clone());
+                }
+            }
+        }
+    }
+    let asset_schema_template = AssetSchema::new(has_aws);
+    let schema_variant_code = asset_schema_template.render()?;
+    let mut mgmt_func_template = RunTemplateMgmtFunc::new(template_components);
+    mgmt_func_template.sort_components_by_dependencies();
+    let mgmt_func_code = mgmt_func_template.render()?;
+    let new_variant = VariantAuthoringClient::create_schema_and_variant_from_code(
+        ctx,
+        asset_name.to_owned(),
+        Some(String::from(r#"## Template Components
+Configure this component to have the Attributes you need to make your templated infrastructure dynamic. The only requirement is that you have a 'domain/Name Prefix' attribute.
+
+Each invocation of the template will create new infrastructure; it's expected that you will have one instance of the template component for potentially many created infrastructures.
+
+Changes in the template are never reflected in the output - components are generated once."#)),
+        None,
+        category,
+        color,
+        schema_variant_code,
+    )
+    .await?;
+    let schema_id = SchemaVariant::schema_id(ctx, new_variant.id()).await?;
+    let func = FuncAuthoringClient::create_new_management_func(
+        ctx,
+        Some(func_name.clone()),
+        new_variant.id(),
+    )
+    .await?;
+    FuncAuthoringClient::update_func(
+        ctx,
+        func.id,
+        Some("Run Template".to_string()),
+        Some("Run the template to create new components".to_string()),
+    )
+    .await?;
+    let prototype_id = ManagementPrototype::list_ids_for_func_id(ctx, func.id)
+        .await?
+        .pop()
+        .ok_or(GenerateTemplateError::FuncMissingPrototype(func.id))?;
+    FuncAuthoringClient::save_code(ctx, func.id, mgmt_func_code).await?;
+    Ok((new_variant, schema_id, func, prototype_id))
+}
+
+fn sanitize_js_variable(input: &str) -> String {
+    static RE: Lazy<Regex> =
+        Lazy::new(|| Regex::new(r"[^a-zA-Z0-9_$]").expect("Regex failed to compile; bug!"));
+    let mut sanitized = RE.replace_all(input, "").to_string();
+
+    if let Some(first_char) = sanitized.chars().next() {
+        if !first_char.is_ascii_alphabetic() && first_char != '_' && first_char != '$' {
+            sanitized.insert(0, '_');
+        }
+    }
+
+    sanitized
+}
```

Comment:
...I'm going to trust you that this works.

Reply:
Just moving the code that was there :)